### DISCUSSION TITLE
Cannot read properties of undefined (reading 'isDragging')

### DISCUSSION THREAD

**Discussion by folarin-vendease** [giovanni-bertoncelli/ngx-konva#26] [Q&A] [answered]
First attempt at using ngx-konva, I am able to render a circle with code below

    import { NgxKonvaModule } from  'ngx-konva';
    import { CommonModule } from  '@angular/common';
    
    @Component({
    
    selector:  'app-floors-and-tables',
    
    templateUrl:  './floors-and-tables.component.html',
    
    styleUrls: ['./floors-and-tables.component.scss'],
    
    standalone:  true,
    
    imports: [CommonModule NgxKonvaModule]
    
    })

---

    <div class="border border-error-500 flex flex-wrap w-[55rem] h-[45rem]">
      <ko-stage [config]="{ width: 860, height: 700 }" [draggable]="true">
        <ko-layer>
          <ko-circle
            *ngFor="let table of tables"
            [config]="{
              radius: 20,
              fill: 'blue',
              stroke: 'black',
              strokeWidth: 1,
              x: table.xPosition,
              y: table.yPosition
            }"
          ></ko-circle>
        </ko-layer>
      </ko-stage>
    </div>

However, upon hovering on any of the `ko-elements`, the error below throws multiple times in the console.

    core.mjs:10614 ERROR TypeError: Cannot read properties of undefined (reading 'isDragging')
        at Object.isDragging (Global.js:46:36)
        at Stage._pointermove (Stage.js:419:28)
        at content.addEventListener.passive (Stage.js:308:33)
        at _ZoneDelegate.invokeTask (zone.js:402:31)
        at core.mjs:10757:55
        at AsyncStackTaggingZoneSpec.onInvokeTask (core.mjs:10757:36)
        at _ZoneDelegate.invokeTask (zone.js:401:60)
        at Object.onInvokeTask (core.mjs:11070:33)
        at _ZoneDelegate.invokeTask (zone.js:401:60)
        at Zone.runTask (zone.js:173:47)
    
  Did I get the setup wrong?
  Please help.

**Comment by ghost** [comment, marked as answer]
Hi, try updating the package to the latest version. There is a regression of konvajs tracked [here](https://github.com/konvajs/konva/issues/1730) in one of their latest versions, I've downgraded to a safer one in the last release.

**Reply by folarin-vendease** [reply]
Thank you!
I currently have these versions installed. Can you recommend what versions to use?

    "konva": "^9.3.6",
    "ngx-konva": "^16.1.5",

**Reply by ghost** [reply]
Yes, take a look to the peerDependencies of the project:
https://github.com/ctinnovation/ngx-konva/blob/main/projects/ngx-konva/package.json

**Reply by folarin-vendease** [reply]
`"konva": "^9.2.3"`
I will try this and give feedback.
Thank you!

**Reply by folarin-vendease** [reply]
The error is no more.
Thank you! @giovanni-bertoncelli

**Reply by ghost** [reply]
@folarin-vendease If you wish to update to the latest `konva` version now with `ngx-konva@16.1.6` you should be able to do so 🥳 .. Let me know if there are any issues!

**Reply by folarin-vendease** [reply]
Thank you for the update! @giovanni-bertoncelli Thank you for the update 🙌
I will keep you posted.
